BST.rb

```diff
@@ -0,0 +1,411 @@
+
+
+class TreeNode
+
+    attr_accessor :value, :left, :right
+
+    def initialize(value = nil, left = nil, right = nil)
+      @value = value
+      @left = left
+      @right = right
+    end
+end
+
+
+class BinarySearchTree
+    
+    def initialize()
+        @root = nil
```

Comment:
Add root as attr_accessor, then no need to do "@root", you can simply use "root".